src/runtime/server/middleware/rate-limit.ts

```diff
@@ -8,17 +8,25 @@ import { useRuntimeConfig } from '#imports'
  * Only works on API routes, eg: /api/hello
  */
 export default defineEventHandler(async (event) => {
-  const isAPI = event.node.req.url?.includes('/api/')
-  const isEnabled = useRuntimeConfig().public.nuxtRateLimit.enabled
+  const { headers } = useRuntimeConfig().nuxtRateLimit
 
-  if (isAPI && isEnabled) {
-    const limited = isRateLimited(event)
+  const payload = getRateLimitPayload(event)
+  // route does not have rate limiting configured
+  if (!payload) {
+    return
+  }
+  const { limited, current, limit, secondsUntilReset } = payload
+
+  if (headers) {
```

Comment:
Opt-in header support can be useful for slowing down requests if reaching api limit